frontend/src/components/maps/hooks/useActiveFeatureLayer.tsx

```diff
@@ -55,7 +61,14 @@ const useActiveFeatureLayer = ({
     let title = 'Municipality Information';
     let feature: Feature | undefined = undefined;
 
-    const parcel = await parcelsService.findOneWhereContains(latLng);
+    // call these APIs in parallel - notice there is no "await"
+    const task1 = parcelsService.findOneWhereContains(latLng);
+    const task2 = regionService.findMetadataByLocation(latLng, 'GEOMETRY');
+    const task3 = districtService.findMetadataByLocation(latLng, 'GEOMETRY');
+
+    const parcel = await task1;
+    const region = await task2;
+    const district = await task3;
```

Comment:
The comment is quite explicit - we prefer to make these requests in parallel, rather than awaiting each task in sequential order